src/gui/src/helpers.js

```diff
@@ -2633,9 +2633,14 @@ window.update_profile = function(username, key_vals){
             const profile = JSON.parse(text);
 
             for (const key in key_vals) {
-                profile[key] = key_vals[key];
-                // update window.user.profile
-                window.user.profile[key] = key_vals[key];
+                if (key_vals[key] === null || key_vals[key] === undefined || key_vals[key] === '') {
```

Comment:
Nit: `null` , `undefined` and `''` are falsy values, which means we can substitute code above by


```suggestion
                if (!!key_vals[key]) {
```

Reply:
resolved